src/Providers/Api/Request/GenericPaymentDataProvider.php

```diff
@@ -212,4 +212,42 @@ public function getStartSessionRequestData(string $paymentCode, Basket $basket)
         $this->validator->validate($requestParams);
         return $requestParams;
     }
+
+    /**
+     * @param string $paymentCode
+     * @param Basket $basket
+     * @return array
+     * @throws \Exception
+     */
+    public function getStartSessionRequestDataFromOrder(string $paymentCode, $order)
+    {
+
+
+        $requestParams = $this->getDefaultPaymentRequestData($paymentCode);
+
+        // Currency not mentioned in API-Doc of Payone
+        $requestParams['currency'] = $order->currency;
+        // amount in smallest unit
+        $requestParams['amount'] = ($order->amount->invoiceTotal) * 100;
```

Comment:
I am not sure what amount should be transferred here. But normally the giftCardAmount should be subtracted from the invoiceTotal to get the amount the customer has to pay.

Reply:
i've added    $requestParams['basket']['couponDiscount'] = $order->amount->giftCardAmount; where couponDiscount is later verified in the code and if it exists it's added to the APi request as voucher object.